### PR TITLE
Add "Run the Docker daemon as a non-root user (Rootless mode)"

### PR DESCRIPTION
**Pull request by AkihiroSuda**
### Proposed changes

Add "Run the Docker daemon as a non-root user (Rootless mode)":
`engine/security/rootless.md`

The content is based on https://github.com/moby/moby/blob/master/docs/rootless.md

`rootless.md` in `moby/moby` will be replaced of the link to
the `docs.docker.com` page compiled from `rootless.md` in this repo.

Signed-off-by: Akihiro Suda <akihiro.suda.cz@hco.ntt.co.jp>

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
cc @thaJeztah

**Comment by AkihiroSuda** [comment]
@thaJeztah @tonistiigi @tiborvass @andrewhsu @tianon PTAL?

File: engine/security/rootless.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+  `/etc/sysctl.d`) and run `sudo sysctl --system`
+
+#### openSUSE
+- `sudo modprobe ip_tables iptable_mangle iptable_nat iptable_filter` is required.
+  This is might be required on other distros as well depending on the configuration.
+
+- Known to work on openSUSE 15.
+
+#### Fedora 31 and later
+- Run `sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"`
```
Comment (thaJeztah) [inline engine/security/rootless.md]: Should we explain what this does? Something like;

```suggestion
- Fedora 31 uses cgroups v2 by default, which is not yet supported by the containerd runtime.
  Run `sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"`
  to use cgroups v1.
```

(open to suggestions there)

File: install/linux/linux-postinstall.md [hunk 2]
```diff
@@ -27,6 +27,13 @@ creates a Unix socket accessible by members of the `docker` group.
 > [*Docker Daemon Attack Surface*](/engine/security/security.md#docker-daemon-attack-surface).
 {: .warning}
 
+> **Note**:
+>
+> To run Docker without the root privileges, see
```
Comment (thaJeztah) [inline install/linux/linux-postinstall.md]:
```suggestion
> To run Docker without root privileges, see
```

File: install/linux/linux-postinstall.md [hunk 3]
```diff
@@ -27,6 +27,13 @@ creates a Unix socket accessible by members of the `docker` group.
...
+> [Run the Docker daemon as a non-root user (Rootless mode)](/engine/security/security.md).
```
Comment (thaJeztah) [inline install/linux/linux-postinstall.md]:
```suggestion
> [Run the Docker daemon as a non-root user (Rootless mode)](/engine/security/rootless.md).
```

File: engine/security/rootless.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+keywords: security, namespaces, rootless
+title: Run the Docker daemon as a non-root user (Rootless mode)
+---
+
+Rootless mode allows running the Docker daemon and containers as a non-root
+user, for the sake of mitigating potentail vulnerabilities in the daemon and
+the container runtime.
+
+Rootless mode does not require the root privileges even for installation of the
+Docker daemon, as long as [the prerequisites](prerequiresites) are satisfied.
```
Comment (thaJeztah) [inline engine/security/rootless.md]:
```suggestion
Docker daemon, as long as [the prerequisites](#prerequisites) are satisfied.
```

**Comment by AkihiroSuda** [comment]
addressed comments, thanks

**Comment by AkihiroSuda** [comment]
@thaJeztah PTAL?

**Comment by thaJeztah** [comment]
@adrian-plata PTAL

File: engine/security/rootless.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+testuser
+$ grep ^$(whoami): /etc/subuid
+testuser:231072:65536
+$ grep ^$(whoami): /etc/subgid
+testuser::231072:65536
+```
+
+### Distribution-specific hint
+
+Using Ubuntu kernel is recommended.
```
Comment (adrian-plata) [inline engine/security/rootless.md]: I would suggest just putting this as a > note instead. So, get rid of the H3 and do something like:

```
> Note: Using Ubuntu kernel is recommended. 
```

**Comment by adrian-plata** [comment]
One suggestion, otherwise LGTM

File: engine/security/rootless.md (hunk 5 above)
Comment (AkihiroSuda) [reply engine/security/rootless.md]: done

**Comment by AkihiroSuda** [comment]
updated

**Pull request merged**